glob.js

```diff
@@ -126,7 +127,9 @@ function Glob (pattern, options, cb) {
     cb = once(cb)
     this.on('error', cb)
     this.on('end', function (matches) {
-      cb(null, matches)
+      setImmediate(function() {
```

Comment:
Can you explain why this is added?  The `end` event should never happen synchronously.